bokeh/document.py

```diff
@@ -244,6 +245,10 @@ def roots(self):
     def title(self):
         return self._title
 
+    @property
+    def id(self):
```

Comment:
I don't love adding this to Document - right now the id is an implementation detail of the embed stuff. When added here, it isn't clear to me how to document what it means. It feels like tunneling an implementation detail through Document.

I'll have to think about cleaner alternatives, not sure yet what to suggest.